code/components/extra-natives-five/src/TrackNatives.cpp

```diff
@@ -146,25 +267,103 @@ static InitFunction initFunction([]()
 	fx::ScriptEngine::RegisterNativeHandler("SET_TRACK_ENABLED", [](fx::ScriptContext& context)
 	{
 		bool state = context.GetArgument<bool>(1);
-		if (rage::CTrainTrack* track = getAndCheckTrack(context, "SET_TRACK_ENABLED"))
+		if (rage::CTrainTrack* track = getAndCheckTrack<0>(context, "SET_TRACK_ENABLED"))
 		{
 			track->m_enabled = state;
 		}
 	});
 
 	fx::ScriptEngine::RegisterNativeHandler("IS_TRACK_ENABLED", [](fx::ScriptContext& context)
 	{
-		if (rage::CTrainTrack* track = getAndCheckTrack(context, "IS_TRACK_ENABLED"))
+		if (rage::CTrainTrack* track = getAndCheckTrack<0>(context, "IS_TRACK_ENABLED"))
 		{
 			context.SetResult<bool>(track->m_enabled);
 		}
 	});
 
 	fx::ScriptEngine::RegisterNativeHandler("IS_TRACK_SWITCHED_OFF", [](fx::ScriptContext& context)
 	{
-		if (rage::CTrainTrack* track = getAndCheckTrack(context, "IS_TRACK_SWITCHED_OFF"))
+		if (rage::CTrainTrack* track = getAndCheckTrack<0>(context, "IS_TRACK_SWITCHED_OFF"))
 		{
 			context.SetResult<bool>(track->m_disableAmbientTrains);
 		}
 	});
+
+	fx::ScriptEngine::RegisterNativeHandler("REGISTER_TRACK_SWITCH", [](fx::ScriptContext& context)
+	{
+		if (rage::CTrainTrack* track = getAndCheckTrack<0>(context, "REGISTER_TRACK_SWITCH"))
+		{
+			int trackNode = context.GetArgument<int>(1);
+			int newTrackNode = context.GetArgument<int>(3);
+
+			if (track->m_nodeCount < trackNode)
+			{
+				fx::scripting::Warningf("natives", "REGISTER_TRACK_SWITCH: Invalid onTrackNode was passed (%i), should be from 0 to %i\n", trackNode, track->m_nodeCount);
+				return;
```

Comment:
Should this not have an "invalid" return type like most other natives?

Reply:
done